exasol_advanced_analytics_framework/interface/create_event_handler_udf.py

```diff
@@ -1,10 +1,162 @@
+import importlib
+from collections import OrderedDict
+from pathlib import PurePosixPath
+from typing import Tuple, Dict, Any, Optional
+from exasol_bucketfs_utils_python.bucketfs_factory import BucketFSFactory
+from exasol_bucketfs_utils_python.bucketfs_location import BucketFSLocation
+from exasol_data_science_utils_python.preprocessing.sql.schema.schema_name import \
+    SchemaName
+from exasol_data_science_utils_python.preprocessing.sql.schema.table_name import \
+    TableName
+
+from exasol_advanced_analytics_framework.event_handler.event_handler_base \
+    import EventHandlerBase
+from exasol_advanced_analytics_framework.event_handler.event_handler_context \
+    import EventHandlerContext
+from exasol_advanced_analytics_framework.event_handler.event_handler_result \
+    import EventHandlerReturnQuery, EventHandlerResultBase
+from exasol_advanced_analytics_framework.event_handler.event_handler_state \
+    import EventHandlerState
+from exasol_advanced_analytics_framework.context_wrapper.udf_context_wrapper \
+    import UDFContextWrapper
+
 
 class CreateEventHandlerUDF:
+
     def __init__(self, exa):
         self.exa = exa
 
     def run(self, ctx) -> None:
-        ctx.emit("return query")
-        ctx.emit("status")
-        ctx.emit("query 1")
-        ctx.emit("query 2")
+        # get and set method parameters
+        iter_num = ctx[0]  # iter_num
+        event_handler_module = ctx[1]  # event_handler_module
+        event_handler_class = ctx[2]  # event_handler_class_name
+        bucketfs_connection = ctx[3]  # bucketfs_connection_name
+        parameters = ctx[4]  # event_handler_parameters
+
+        bucketfs_location = BucketFSFactory().create_bucketfs_location(
+            url=bucketfs_connection.address,
+            user=bucketfs_connection.user,
+            pwd=bucketfs_connection.password,
+            base_path=parameters["base"]
+        )
+        latest_bucketfs_path = PurePosixPath(
+            f"{event_handler_class}_{str(iter_num)}.pkl")
+
+        # load the latest (create if not) event handler state object
+        latest_state = self._load_latest_state(
+            iter_num, event_handler_class, event_handler_module, parameters,
+            bucketfs_location, latest_bucketfs_path)
+        event_handler_context: EventHandlerContext = latest_state.context
+        event_handler: EventHandlerBase = latest_state.event_handler
+
+        # call the user code
+        udf_context = self._create_udf_context_wrapper(ctx)
+        result: EventHandlerResultBase = event_handler.handle_event(
+            udf_context, event_handler_context)
+
+        # save the current state
+        iter_num += 1
+        current_bucketfs_path = PurePosixPath(
+            f"{event_handler_class}_{str(iter_num)}.pkl")
+        current_state = EventHandlerState(event_handler_context, event_handler)
```

Comment:
you need to store the columns of the return_query, such that you can use them in the next iteration for creating the UDFContext

Reply:
In addition, even_handler_context is already part of the event_handler object. It will be redundant to keep in state object.

Reply:
But event_handler_context doesn't store the columns of the return query and no one else does it, that means you need to add them EventHandlerState. I think that is also where they belong to. They are not part of the event_handler_context, because columns of the return query are only for the next iteration and the event_handler_context exists over the whole lifetime of the event handler.

Reply:
No, you misunderstood me. We have to add retrun_query columns, there is no problem with that. I will update what I mean in the next PR

Reply:
ok, I now understand your sentence, but I disagree, the event_handler_context is not part of the event_handler, we only give it via the handle_next. We could think, about injecting via the constructor of the event_hanlder. This, however, has the risks, because we can't replace it through the iterations, if the implementation would make it necessary.